### PR TITLE
Zephyr 64-bit port: initial set of changes

### PR DESCRIPTION
**Pull request by npitre**
This is my first set of patches containing very simple changes to start
with. Those should be pretty uncontrovertial. Also included is a genuine
bugfix for the PTP test code. Much more is coming after this.

### CODE REVIEW

File: boards/posix/native_posix/irq_ctrl.h [hunk 1]
```diff
@@ -26,7 +26,7 @@ uint8_t hw_irq_ctrl_get_prio(unsigned int irq);
 int hw_irq_ctrl_get_highest_prio_irq(void);
 uint32_t hw_irq_ctrl_get_current_lock(void);
 uint32_t hw_irq_ctrl_change_lock(uint32_t new_lock);
```
Comment (tbursztyka) [inline boards/posix/native_posix/irq_ctrl.h]: looks like these 2 above have the same issue s/uint32_t/u32_t

File: tests/net/ptp/clock/src/main.c [hunk 2]
```diff
@@ -175,29 +175,40 @@ static u64_t timestamp_to_nsec(struct net_ptp_time *ts)
 	return (ts->second * NSEC_PER_SEC) + ts->nanosecond;
 }
 
+struct ptp_context {
```
Comment (tbursztyka) [inline tests/net/ptp/clock/src/main.c]: how does this patch relate to 64 bits support? (commit title is wrong btw, should be "tests/net: Fix PTP test case" for instance)

**Comment by npitre** [comment]
On Tue, 28 May 2019, Tomasz Bursztyka wrote:

> +struct ptp_context {
> 
> how does this patch relate to 64 bits support? (commit title is wrong btw, should be "tests/net: Fix PTP test case" for instance)

Related in the sense that it triggers a crash for me only on a 64-bit 
build.

I'm fixing titles and so to please the police bot.

**Comment by andyross** [comment]
Heh, was about to +1 this when I noticed you'd closed it.

Regardless, and notwithstanding the formatting and style nitpicks which I'll leave to others, this looks great to me.  I particularly love the fact that it's happening in native_posix and not x86_64.  In theory we can just turn it on there when this is proven out.

**Comment by jukkar** [comment]
Good catch with the ptp_clock stuff. I think you could send a separate PR for just that fix as it is not really related to 64-bit support.

**Comment by aescolar** [comment]
Replaced by #16461

**Comment by npitre** [comment]
On Wed, 29 May 2019, Jukka Rissanen wrote:

> Good catch with the ptp_clock stuff. I think you could send a separate PR for just that fix as it is not really related to 64-bit support.

Done.

**Pull request closed without merging**